java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/ingest/RunWriteRandomDataTaskOnECSForMultipleNewTables.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.systemtest.drivers.ingest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.ecs.EcsClient;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import sleeper.clients.table.AddTable;
+import sleeper.clients.table.TakeAllTablesOffline;
+import sleeper.configuration.properties.S3InstanceProperties;
+import sleeper.configuration.properties.S3TableProperties;
+import sleeper.core.properties.PropertiesUtils;
+import sleeper.core.properties.instance.CommonProperty;
+import sleeper.core.properties.instance.InstanceProperties;
+import sleeper.core.properties.table.TableProperties;
+import sleeper.core.properties.table.TablePropertiesStore;
+import sleeper.core.properties.table.TableProperty;
+import sleeper.core.schema.Schema;
+import sleeper.core.statestore.StateStoreProvider;
+import sleeper.statestore.StateStoreFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static sleeper.configuration.utils.AwsV2ClientHelper.buildAwsV2Client;
+
+public class RunWriteRandomDataTaskOnECSForMultipleNewTables {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RunWriteRandomDataTaskOnECSForMultipleNewTables.class);
+
+    private S3Client s3Client;
+    private DynamoDbClient dynamoClient;
+    private EcsClient ecsClient;
+
+    public RunWriteRandomDataTaskOnECSForMultipleNewTables(S3Client s3Client, DynamoDbClient dynamoClient, EcsClient ecsClient) {
+        this.s3Client = s3Client;
+        this.dynamoClient = dynamoClient;
+        this.ecsClient = ecsClient;
+    }
+
+    public void takeAllTablesOffline(String instanceId) {
+        TakeAllTablesOffline offliner = new TakeAllTablesOffline(this.s3Client, this.dynamoClient);
```

Comment:
There are unnecessary uses of "this" in this file. Please see our coding conventions:

https://github.com/gchq/sleeper/blob/develop/docs/development/conventions.md#use-of-this